walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -149,4 +138,64 @@ private static void searchForNextEvent(List<Event> eventList, TimeRange currEven
   private static boolean requestFits(MeetingRequest request, int nextEventStart, int currEventEnd) {
     return request.getDuration() <= nextEventStart - currEventEnd;
   }
+
+  private static void handleBeginningOfDay(List<TimeRange> availableTimes, List<Event> eventList,
+    MeetingRequest request) {
+    Collections.sort(eventList, Event.ORDER_BY_START);
+    TimeRange firstEventTimeRange = eventList.get(0).getWhen();
+    
+    // Add the first time slot, from the start of the day to the start of the first event,
+    // only if there's no event that starts the day.
+    if (startOfDayIsFree(eventList) && requestFits(request, firstEventTimeRange.start(), TimeRange.START_OF_DAY)) {
+      availableTimes.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY,
+        firstEventTimeRange.start(), false));
+    }
+  }
+  
+  /**
+   * Go through each event and check if it overlaps with another. If it does, we connect
+   * the end of our event with the start of the next one that is not overlapped. Otherwise,
+   * just connect the 2 events to create a time slot, if the request fits.
+   */
+  private static void handleMiddleOfDay(List<TimeRange> availableTimes, List<Event> eventList,
+    MeetingRequest request) {
+    for (int i = 0; i < eventList.size() - 1; i++) {
+      TimeRange currEventTimeRange = eventList.get(i).getWhen();
+      TimeRange nextEventTimeRange = eventList.get(i + 1).getWhen();
+      if (eventsOverlap(currEventTimeRange, nextEventTimeRange)){
+        removeOverlappedEvents(i, eventList, currEventTimeRange); 
+        searchForNextEvent(eventList, currEventTimeRange, i, availableTimes, request);
+      } else if (requestFits(request, nextEventTimeRange.start(), currEventTimeRange.end())) {
+          availableTimes.add(TimeRange.fromStartEnd(currEventTimeRange.end(),
+            nextEventTimeRange.start(), false));  
+        }
+    }
+  }
+
+  private static void handleEndOfDay(List<TimeRange> availableTimes, List<Event> eventList,
+    MeetingRequest request) {
+    // Sort them again to be able to get the last end. Might be not that efficient.
+    Collections.sort(eventList, Event.ORDER_BY_END);
+    TimeRange lastEventTimeRange = eventList.get(eventList.size() - 1).getWhen();
+    
+    // Handle the case where the meeting can take place at the end of the day
+    if (!availableTimes.isEmpty() && requestFits(request, TimeRange.END_OF_DAY, lastEventTimeRange.end())) {
+      if (nothingEndsTheDay(availableTimes, eventList)) {
+        availableTimes.add(TimeRange.fromStartEnd(lastEventTimeRange.end(),
+          TimeRange.END_OF_DAY, true));
+      } 
+    } else if (fitsOnlyAtTheEnd(request, availableTimes, eventList)) {
+        availableTimes.add(TimeRange.fromStartEnd(lastEventTimeRange.end(),
+          TimeRange.END_OF_DAY, true));
+      }
+  }
+
+  private static void removeOverlappedEvents(int currEventPosition, List<Event> eventList,
```

Comment:
please document this method, highlighting why we start from `currEventPosition + 2`